packages/runtime/src/schema.ts

```diff
@@ -377,15 +391,66 @@ class SchemaBuilder {
 
   _buildObjectType(params: BuildObjectParams) {
     const model = this.#def.models[params.name]
-    return model
-      ? this._buildDocumentObjectType({ model, ...params })
-      : this._buildEmbeddedObjectType(params)
+    if (model == null) {
+      this._buildEmbeddedObjectType(params)
+    } else {
+      const modelParams = { model, ...params }
+      if (model.interface) {
+        this._buildInterfaceObjectType(modelParams)
+      } else {
+        this._buildDocumentObjectType(modelParams)
+      }
+    }
+  }
+
+  _buildInterfaceObjectType({ model, definitions, name, fields }: BuildModelObjectParams) {
+    this.#types[name] = new GraphQLInterfaceType({
+      name,
+      interfaces: () => this._resolveInterfaces(name, model.implements),
+      fields: () => {
+        const config: GraphQLFieldConfigMap<ModelInstanceDocument, Context> = {}
+        for (const [key, field] of Object.entries(fields)) {
+          switch (field.type) {
+            case 'meta':
+              // Don't show meta fields in schema
+              continue
+            case 'reference':
+              config[key] = this._buildDocumentObjectReferenceField(key, field)
+              break
+            case 'list':
+              config[key] = {
+                type: this._buildObjectListFieldType(definitions, field),
+                resolve: (doc) => doc.content?.[key] as unknown,
+              }
+              break
+            case 'view': {
+              const view = this._buildDocumentObjectViewField(key, definitions, field, fields)
+              if (view) {
+                config[key] = view
+              }
+              break
+            }
+            default:
+              config[key] = {
+                type: this._buildScalarFieldType(definitions, field),
+                resolve: (doc) => doc.content?.[key] as unknown,
+              }
+          }
+        }
+        return config
+      },
+    })
+
+    this._buildFiltersType(name, fields)
+    this._buildSortingType(name, fields)
   }
 
-  _buildDocumentObjectType({ model, definitions, name, fields }: BuildDocumentObjectParams) {
+  _buildDocumentObjectType({ model, definitions, name, fields }: BuildModelObjectParams) {
     this.#types[name] = new GraphQLObjectType<ModelInstanceDocument>({
       name,
-      interfaces: [definitions.nodeInterface],
+      interfaces: () => {
```

Comment:
Does this need to be a function, seems like it was just an array before?

Reply:
Yes, that allows to defer the resolution until after all the types are defined